spoonbill/utils.py

```diff
@@ -165,14 +173,23 @@ def generate_table_name(parent_table, parent_key, key):
     >>> generate_table_name('tenders', 'tender', 'items')
     'tenders_items'
     >>> generate_table_name('tenders', 'items', 'additionalClassifications')
-    'tenders_items_addit'
+    'tenders_items_class'
     >>> generate_table_name('parties', 'parties', 'roles')
     'parties_roles'
     """
+
+    if key in ABBREVIATIONS:
+        key = ABBREVIATIONS[key]
+
     if parent_key in parent_table:
-        return f"{parent_table}_{key[:5]}"
+        table_name = f"{parent_table}_{key[:5]}"
```

Comment:
probably shortening needed only when total length of table name is longer than 31 character